lambda/fulfillment/lib/middleware/lex.js

```diff
@@ -131,7 +131,15 @@ function slackifyResponse(response) {
         let md = response.result.alt.markdown;
         qnabot.log("Converting markdown response to Slack format.");
         qnabot.log("Original markdown: ", JSON.stringify(md));
+
+        md = md.replace(/<\/?span[^>]*>/g,"");  // remove any span tags (eg no-translate tags)
+        md = md.replace(/<\/?br *\/?>/g,"\n"); // replace br with \n
+
         md = slackifyMarkdown(md);
+
+        //decode URIs in markdown -- slackify-markdown encodes URI. If presented with an encoded URI, slackify-markdown is double encoding URIs
+        md = decodeURI (md); 
```

Comment:
Did you verify if this presents any issues if the md value contains a unencoded URL?  By slackifying and then decoding, unencoded URL will remain unencoded.. is tat OK when using the slack channel?

Reply:
Tried with a variety of combinations of URLs - with and without encoding. So far, the fix seems to hold okay. We should test more to make sure we are not missing anything.